sympy/simplify/powsimp.py

```diff
@@ -366,7 +366,8 @@ def update(b):
             c_powers.append((b, e))
         check = len(c_powers)
         c_powers = dict(c_powers)
-        assert len(c_powers) == check  # there should have been no duplicates
+        if len(c_powers) != check:  # there should have been no duplicates
+            raise ValueError("Unexpected duplicates in c_powers")
```

Comment:
Many of these asserts should remain as asserts. I'm not going to go through them all one by one but there are good reasons for using assert rather than raising ValueError.

The point of an assert is that it should never be raised unless there is a bug. There is no reason to raise ValueError because no one should ever need to catch the exception.

Changing the assert to `if ...: raise ValueError` subtly changes the interpretation of the code from "this condition should always be false" to "there may be some reasonable cases where this is false" which is an important difference when reading the code.